core/application/notificationWorkflow/NotificationService.ts

```diff
@@ -1,54 +1,122 @@
 import { GENERIC_CODES } from '../../../commons/libs/constants/GenericCodes'
 import { UserDetailsDTO } from '../../../commons/dto/UserDTO'
-import { NotificationDTO, NotificationStatus } from '../../../commons/dto/NotificationDTO'
+import {
+  BloodDonationNotificationDTO,
+  NotificationDTO,
+  NotificationStatus
+} from '../../../commons/dto/NotificationDTO'
 import NotificationOperationError from './NotificationOperationError'
-import { BloodPostNotificationAttributes, NotificationAttributes, SnsRegistrationAttributes, StoreNotificationEndPoint } from './Types'
+import {
+  BloodDonationNotificationAttributes,
+  BloodDonationPayloadAttributes,
+  NotificationAttributes,
+  SnsRegistrationAttributes,
+  StoreNotificationEndPoint
+} from './Types'
 import Repository from '../models/policies/repositories/Repository'
 import { SNSModel } from '../../application/models/sns/SNSModel'
 import { generateUniqueID } from '../utils/idGenerator'
 import { QueueModel } from '../models/queue/QueueModel'
+import NotificationRepository from '../models/policies/repositories/NotificationRepository'
 
 export class NotificationService {
   async publishNotification(
     notificationAttributes: NotificationAttributes,
     userSnsEndpointArn: string,
-    notificationRepository: Repository<NotificationDTO>,
     snsModel: SNSModel
   ): Promise<string> {
     try {
-      const { userId, type, payload } = notificationAttributes
+      await snsModel.publish(notificationAttributes, userSnsEndpointArn)
+      return 'Notified user successfully.'
```

Comment:
why does need to return a string?

Reply:
Updated in 77a84e14b8a0b4fe53ac3803ea7b36243c3810f7